docs/modules/ROOT/pages/reference/extensions/ldap.adoc

```diff
@@ -39,3 +43,149 @@ Please refer to the above link for usage and configuration details.
 ifeval::[{doc-show-user-guide-link} == true]
 Check the xref:user-guide/index.adoc[User guide] for more information about writing Camel Quarkus applications.
 endif::[]
+
+[id="extensions-ldap-usage"]
+== Usage
+[id="extensions-ldap-usage-dircontext"]
+=== DirContext
+
+The URI, `ldap:ldapserver`, references a bean with the ID `ldapserver`. A CDI producer method may be used to instantiate a `DirContext` object as follows:
+
+[source,java]
+----
+public class LdapServerProducer {
+
+    @Produces
+    @Dependent
+    @Named("ldapserver")
+    public DirContext createLdapServer() throws Exception {
+        Hashtable<String, String> env = new Hashtable<>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        env.put(Context.PROVIDER_URL, "ldap://localhost:10389");
+        env.put(Context.SECURITY_AUTHENTICATION, "none");
+
+        return new InitialDirContext(env);
+    }
+}
+----
+
+The preceding example creates a regular Sun based LDAP `DirContext` that connects anonymously to a locally hosted LDAP server. The use of the `@Named` annotation binds the `DirContext` into the Camel registry automatically.
+
+[id="extensions-ldap-usage-configuring-ssl"]
+=== Configuring SSL
+
+When connecting to an LDAP server over SSL/TLS, you may encounter situations where the default trust manager used by the JVM is unable to verify the certificate. This can happen, for example, when the server uses a self-signed certificate or when the certificate is issued by a non-trusted CA. In such cases, you may need to provide a custom trust manager implementation that can verify the server's certificate.
+
+The following code shows an implementation of a custom socket factory that can be used to create SSL/TLS sockets. The class name of the custom SSL socket factory is then specified in the `java.naming.ldap.factory.socket` property of the environment hashtable used to create the LDAP context.
+
+[source,java]
+----
+public class CustomSSLSocketFactory extends SSLSocketFactory {
+
+    private SSLSocketFactory delegate;
+
+    public CustomSSLSocketFactory() throws Exception {
+        String trustStoreFilename = ConfigProvider.getConfig().getValue("ldap.trustStore", String.class);
+        String trustStorePassword = ConfigProvider.getConfig().getValue("ldap.trustStorePassword", String.class);
+        KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        try (InputStream in = new FileInputStream(trustStoreFilename)) {
+            keyStore.load(in, trustStorePassword.toCharArray());
+        }
+        TrustManagerFactory tmf = TrustManagerFactory.getInstance("SunX509");
+        tmf.init(keyStore);
+        SSLContext ctx = SSLContext.getInstance("TLS");
+        ctx.init(null, tmf.getTrustManagers(), null);
+        delegate = ctx.getSocketFactory();
+
+    }
+
+    public static SocketFactory getDefault() {
```

Comment:
Do we need this?

If we do, maybe it'd be better to add whatever exceptions are thrown to the method signature instead of printing stack traces & returning `null`.

Reply:
I've just had a play around and it looks like `getDefault()` is needed. It is called by the LdapCtxFactory. The method itself is defined on the parent `SSLSocketFactory`, so I can't change the signature.
I'm not really sure what the best way is to handle exceptions. The example included in the [Camel docs](https://camel.apache.org/components/next/ldap-component.html#_configuring_ssl) essentially moves the try/catch block to the constructor. The _Using Custom Sockets_ section of the official [Java SSL](https://docs.oracle.com/javase/jndi/tutorial/ldap/security/ssl.html) guide ignores the prospect of exceptions entirely.